### PR TITLE
"eosio_uninstall.sh --full" to delete directories with configs

### PR DESCRIPTION
**Pull request by NorseGaud**
Adding code to support --full for eosio_uninstall.sh to remove directories with configs.

- I'm sure there are more directories, please add whatever I haven't
- I would like to add an uninstall guide under "BUILDING EOSIO" on https://developers.eos.io/eosio-nodeos/docs. Can someone add this page for me and I'll suggest the edits?

Thank you

### CODE REVIEW

File: eosio_build.sh [hunk 1]
```diff
@@ -30,6 +30,11 @@
 # https://github.com/EOSIO/eos/blob/master/LICENSE
 ##########################################################################
 
+if [ "$(id -u)" -ne 0 ]; then
+	printf "\n\tThis requires sudo. Please run with sudo.\n\n"
+	exit -1
+fi
```
Comment (heifner) [inline eosio_build.sh]: eosio_build.sh should not require sudo

**Comment by NorseGaud** [comment]
Curious why you didn't edit and remove the sudo requirement, @heifner?

**Comment by heifner** [comment]
@NorseGaud it was a bit of a fly-by review. Just not a high priority at the moment. Thanks for the contribution.

**Comment by NorseGaud** [comment]
@heifner Appreciate the follow-up. Just curious if there is a process I should be following to contribute. Want to follow the guidelines if they exist and make your life easier. Thanks

**Comment by heifner** [comment]
Actually, PRs should be against `release/1.2.x` if bug fix OR `develop` if new feature. `develop` if unsure.

**Comment by heifner** [comment]
As for docs. Pick a close page and suggest edits there.

File: eosio_uninstall.sh [hunk 2]
```diff
@@ -26,9 +26,15 @@ if [ -d "/usr/local/eosio" ]; then
             for binary in ${binaries[@]}; do
                rm ${binary}
             done
+            # Handle cleanup of directories created from installation
+            if [ "$1" == "--full" ]; then
+               if [ -d ~/Library/Application\ Support/eosio ]; then rm -rf ~/Library/Application\ Support/eosio; fi # Mac OS
+               if [ -d ~/.local/share/eosio ]; then rm -rf ~/.local/share/eosio; fi # Linux
+               if [ -d /usr/local/eosio ]; then rm -rf /usr/local/eosio; fi
```
[the remark this answers is not in the document]
Comment (NorseGaud) [reply eosio_uninstall.sh]: You're right, good catch. Thanks!

**Comment by NorseGaud** [comment]
All set I think. Any updates for this?

**Pull request merged**